scripts/fix_sq_headers.py

```diff
@@ -138,23 +157,33 @@ def do_reheader(dry_run, refset, newref, newrefset, incram, outcram, outcrai):
     }
 
 
-def do_metamist_update(dataset, sequencing_type, seqgroup, oldpath, newpath, result):
+def do_metamist_update(dataset, activeseqgroup, oldanalysis, oldpath, newpath, result):
     aapi = AnalysisApi()
 
-    analysis = models.Analysis(
-        type='cram',
-        status=models.AnalysisStatus('completed'),
+    newmeta = oldanalysis['meta']
+    newmeta['size'] = result['cram_size']
+    newmeta['source'] += f' (reheadered from {oldpath})'
+
+    newanalysis = models.Analysis(
+        type=oldanalysis['type'],
+        status=models.AnalysisStatus(oldanalysis['status'].lower()),
         output=str(newpath),
-        sequencing_group_ids=[seqgroup],
-        meta={
-            'sequencing_type': sequencing_type,
-            'size': result['cram_size'],
-            'source': f'Reheadered from {oldpath}',
-        },
+        sequencing_group_ids=[activeseqgroup['id']],
+        meta=newmeta,
     )
-    aid = aapi.create_analysis(dataset, analysis)
+    aid = aapi.create_analysis(dataset, newanalysis)
     print(f'Created Analysis(id={aid}, output={newpath}) in {dataset}')
 
+    for analysis in activeseqgroup['analyses']:
+        activeid = analysis['id']
+        analysisupdate = models.AnalysisUpdateModel(
+            active=False,
+            status=models.AnalysisStatus(analysis['status'].lower()),
+        )
+        if not aapi.update_analysis(activeid, analysisupdate):
```

Comment:
would it be worth putting a small wait time in here to avoid hitting the metamist server a bunch of times in a loop without any let up? Or is the amount of the analyses on the seq group expected to be small?

Reply:
To be honest, the number of analyses on the seq group is expected to be 1 but I put the loop in just to make sure!

Reply:
No worries, that was my guess but just wanted to double check, all good with me then